File: Searching and Sorting/Linear Search/Java/LinearSearch.java
import java.util.Scanner;
class LinearSearch
{
   public static void main(String args[])
   {
      int counter, numOfElements, itemToSearch, array[];
      Scanner input = new Scanner(System.in);
      System.out.println("Enter number of elements:");
      numOfElements = input.nextInt(); 
      array = new int[numOfElements]; 
      System.out.println("Enter elements");
      for (counter = 0; counter < numOfElements; counter++)
        array[counter] = input.nextInt();
      System.out.println("Enter the search value:");
      itemToSearch = input.nextInt();
      for (counter = 0; counter < numOfElements; counter++)
      {
         if (array[counter] == itemToSearch) 
         {
           System.out.println(itemToSearch+" is present at  "+(counter+1));
           break;
         }
      }
      if (counter == numOfElements)
        System.out.println(itemToSearch + " is not present");
   }
}

File: Searching and Sorting/Linear Search/C/Linear_search.c
/* In Linear search, we search an element or value in a given array by traversing the array from the starting,
till the desired element or value is found.
If the desired element is found, the index position is displayed, else -1 is displayed.*/

#include<stdio.h>

int linear_search(int* arr, int ele, int n);
// function returns the index position if the element is found, else return -1

int main()
{
    int n;
    printf("Enter the number of elements: ");
    scanf("%d",&n);

    int arr[n];
    printf("Enter the elements of the array: \n");
    for(int i = 0; i<n; i++)
        scanf("%d",&arr[i]);

    int ele;
    printf("Enter the element to be searched: ");
    scanf("%d",&ele);

    int pos = linear_search(arr,ele,n);

    if(pos==-1) printf("%d not found.",ele);
    else printf("%d found at index position %d",ele,pos);

    return 0;
}


int linear_search(int* arr, int ele, int n)
{
    for(int i = 0; i<n; i++)
        if(arr[i]==ele) return i;

    return -1;
}